railties/test/generators/app_generator_test.rb

```diff
@@ -831,6 +840,14 @@ def test_edge_option
     assert_file "Gemfile", %r{^gem\s+["']rails["'],\s+github:\s+["']#{Regexp.escape("rails/rails")}["']$}
   end
 
+
+  def test_master_option
+    generator([destination_root], master: true, skip_webpack_install: true)
+
+    assert_bundler_command_called("install")
+    assert_file "Gemfile", %r{^gem\s+["']rails["'],\s+github:\s+["']#{Regexp.escape("rails/rails")}["'],\s+branch:\s+["']master["']$}
```

Comment:
@roramirez thank you so much for doing this! As I don't have a tremendous amount of experience writing Rails tests in this area, I wasn't sure where to start or jump in. Now that I see your approach that will definitely help me out in the future. Thank you! 🙏

Reply:
Your welcome! Me neither :nerd_face: 

It made sense to me sense we'll should need to cover this area instead try by hand testing.

btw!, thanks for the feature is really help when you need try new things and living in the edge for development.